libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/route/FasterRouteDetector.java

```diff
@@ -0,0 +1,133 @@
+package com.mapbox.services.android.navigation.v5.route;
+
+import android.location.Location;
+
+import com.mapbox.api.directions.v5.models.DirectionsResponse;
+import com.mapbox.api.directions.v5.models.DirectionsRoute;
+import com.mapbox.api.directions.v5.models.LegStep;
+import com.mapbox.api.directions.v5.models.RouteLeg;
+import com.mapbox.services.android.navigation.v5.routeprogress.RouteProgress;
+import com.mapbox.services.android.navigation.v5.routeprogress.RouteStepProgress;
+import com.mapbox.services.android.navigation.v5.utils.time.TimeUtils;
+
+import java.util.Date;
+import java.util.concurrent.TimeUnit;
+
+import static com.mapbox.services.android.navigation.v5.navigation.NavigationConstants.NAVIGATION_CHECK_FASTER_ROUTE_INTERVAL;
+import static com.mapbox.services.android.navigation.v5.navigation.NavigationConstants.NAVIGATION_MEDIUM_ALERT_DURATION;
+
+public class FasterRouteDetector extends FasterRoute {
+
+  private static final int VALID_ROUTE_DURATION_REMAINING = 600;
+
+  private Location lastCheckedLocation;
+
+  @Override
+  public boolean shouldCheckFasterRoute(Location location, RouteProgress routeProgress) {
+    if (location == null || routeProgress == null) {
+      return false;
+    }
+    // On first pass through detector, last checked location will be null
+    if (lastCheckedLocation == null) {
+      lastCheckedLocation = location;
+    }
+    // Check if the faster route time interval has been exceeded
+    if (secondsSinceLastCheck(location) > NAVIGATION_CHECK_FASTER_ROUTE_INTERVAL) {
+      lastCheckedLocation = location;
+      // Check for both valid route and step durations remaining
+      if (validRouteDurationRemaining(routeProgress) && validStepDurationRemaining(routeProgress)) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  @Override
+  public boolean isFasterRoute(DirectionsResponse response, RouteProgress routeProgress) {
+    if (validRouteResponse(response)) {
+
+      double currentDurationRemaining = routeProgress.durationRemaining();
+      DirectionsRoute newRoute = response.routes().get(0);
+
+      if (hasLegs(newRoute)) {
```

Comment:
It looks like this check is missing https://github.com/mapbox/mapbox-navigation-ios/blob/master/MapboxCoreNavigation/RouteController.swift#L669

Reply:
I'm doing that check down on line `60` with https://github.com/mapbox/mapbox-navigation-android/pull/638/files#diff-eda094b0008667fdddf8a3bc30e37bd2R101